cookbooks/zarr-stac-report/data-consumers/index.qmd

```diff
@@ -0,0 +1,98 @@
+
+Constructing the lazy data cube is likely to be very fast if there is a [consolidated metadata file][9] OR the data is in Zarr-3 and the Zarr metadata fetch is highly parallelized ([read more][4]).
+
+
+## Many small Zarr stores in STAC items
+
+The STAC catalog contains many items in each collection and the items each contain one asset pointing to a Zarr store. In this setup, the Zarr store represents a particular area in time and space (a scene). This is very similar to the common practice of using COGs and STAC where each COG represents one band at one scene. The difference is instead of one band per COG, all bands are in one Zarr store (this is conceptually equivalent to a multi-band COG).
+
+In this scenario the STAC metadata can be used to find data that overlaps the region of interest (both in time and space) before constructing a lazily-loaded data cube in xarray.
+
+### Filter in STAC first
+
+If you are interacting with a STAC API and the Zarr store is cataloged in STAC following [`xstac`][10] conventions then all the metadata you need for filtering and subsetting is available at the STAC level so you should be able to search either in the web-browser directly or using pystac-client.
+
+If you are interacting with a static STAC catalog (in json or geoparquet) you might be able to use [rustac][3] and [stac-geoparquet][2] to search the collection and find the items of interest. _This is a newer approach and might have rough edges_.
+
+In both the API and static STAC case the result of the search will be an iterable of items. Each of these items will have at least one asset pointing to the Zarr store (sometimes there will be multiple assets that point to the same store using different protocols). Take the assets that meet your needs and you can use `xr.open_mfdataset` with the `zarr` or `stac` backend to construct the data cube. 
+
+If the Zarr stores all cover the same area and are on the same grid then concatenation might be straightforward. If you need to apply some transformation (such as regridding or doing a projection) before concatenating the data in the files then you might find it easier to use  `xr.open_dataset` directly and iterate over the items in the list applying transformations before using one of xarray's combine methods.
+
+If the Zarr store includes data at several resolutions (for instance if it has overviews) then it might make more sense to open it as an [xarray.DataTree][8] and pick out the arrays that are of interest before potentially reprojecting and combining across the time dimension.
+
+## Virtual reference files in STAC
+
+Virtual reference files can be treated exactly like how you would treat a Zarr store. They can be lazily-loaded into xarray using the [`kerchunk`][5] or [`stac`][1] backend and doing so will result in one GET request containing the minimal metadata.
```

Comment:
Should we also include icechunk here?

Reply:
I don't really have ideas yet about how to refer to icechunk from STAC. Maybe that can be added later on?